arch/arm/core/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -103,7 +103,7 @@ static inline void _get_mpu_ram_region_attr(arm_mpu_region_attr_t *p_attr,
 	 */
 	(void) base;
 
-	p_attr->rasr = _get_region_attr(1, ap, 1, 1, 1, 1, 0, size);
+	p_attr->rasr = _get_region_attr(1, ap, 1, 1, 1, 0, 0, size);
```

Comment:
Quick comment: this patch will affect the MPU of all ARMv6-M/v7-M variants (M0+, M3, M4, M7). So we need to justify its need for all cases, not only for Cortex-M7.

Reply:
Otherwise, I have no objections to the patch; making the thread/application memory regions non-shareable.

Reply:
I have browsed the Cortex M0+, M3 and M4 documentation, and I haven't found any of comment about the cache behaviour for shareable regions. On the other hand theses cores do not have cache.

I think one important thing there is to have a consistent definition of the cache attributes in the various cases and avoid having a region which uses different cache attributes depending if the memory is accessed from the thread/application or from the kernel. Zephyr currently marks all the SRAM as non-shareable except the thread/application memory regions. This looks a strange choice to me.

Reply:
> Zephyr currently marks all the SRAM as non-shareable except the thread/application memory regions. This looks a strange choice to me.

It might well have been an unintentional configuration. I do not recall specific reasons why the thread and application memory stacks would need to be sharable.